### PR TITLE
updated developers documentation to include k2-tools and references

### PR DESCRIPTION
**Pull request by alejandroEsc**

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -97,6 +97,46 @@ GitHub.
     3. K2 committers will iterate with you on the design to make sure you are on the right track.
     4. Implement your issue, create a pull request (see below), and iterate from there.
 
+
+### Recommended Dev Tooling 
+The recommended K2 Developer Tool can be found in the `/hack` directory and is called [k2-tools](https://github.com/samsung-cnct/k2-tools), the intent is to mitigate version issues and requirements,
```
Comment (coffeepac) [inline DEVELOPERS.md]: is this link intended to be 'dockerdev'?  k2-tools doesn't exist in that directory.

File: DEVELOPERS.md [hunk 2]
```diff
@@ -97,6 +97,46 @@ GitHub.
...
+allowing you the Developer to focus on coding and not worry on dependencies.
+
+#### Developer Workflow with K2-Tools
+Here we document how a developer would use k2-tools to work with a cluster.
+
+##### Assumptions
+* You have used K2 or k2cli to generate a config file.
+* You have pulled the latest k2 image on your machine (keeps you on track with changes upstream)
```
Comment (coffeepac) [inline DEVELOPERS.md]: ~~image~~ master branch

or

~~k2~~ k2-tools

File: DEVELOPERS.md (hunk 1 above)
Comment (alejandroEsc) [inline DEVELOPERS.md]: hmm, a bit of context switching here, yes i see that it is somewhat ambiguous. I'll clean it up, as its more intended to point to the documentation for k2tools that this tools pulls. Ill try to make that more clear.

File: DEVELOPERS.md (hunk 2 above)
Comment (alejandroEsc) [inline DEVELOPERS.md]: darn, good catch, k2 master branch.

**Comment by coffeepac** [comment]
this is great.  but @alejandroEsc you'll need to rebase onto master to pass CI.  Some critical changes have landed in the last few days.

**Comment by alejandroEsc** [comment]
Understood, will do.

**Comment by alejandroEsc** [comment]
rebased to master, still failed, next steps?

**Comment by coffeepac** [comment]
PR job failed due to a known issue with jenkins.  I have restarted the job.

**Comment by coffeepac** [comment]
@alejandroEsc if you can, please rebase onto master again.  There is a fairly important CI stability fix that went in this morning.

**Comment by alejandroEsc** [comment]
@coffeepac ready to be merged :)

**Pull request merged**